### PR TITLE
Fix for upstream changes

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

File: src/format.rs [hunk 1]
```diff
@@ -72,14 +72,14 @@ pub struct UrlNoFragmentFormatter<'a> {
...
+        try!(formatter.write_str(&*self.url.scheme));
+        try!(formatter.write_str(":"));
         try!(self.url.scheme_data.fmt(formatter));
         match self.url.query {
             None => (),
             Some(ref query) => {
-                try!(formatter.write(b"?"));
-                try!(formatter.write(query.as_bytes()));
+                try!(formatter.write_str("?"));
+                try!(formatter.write_str(&**query));
```
Comment (SimonSapin) [inline src/format.rs]: Is `&**query` preferable to `query.as_slice()`? (Same for `&*self.url.scheme` v.s. `self.url.scheme.as_slice()`)

Comment (sfackler) [inline src/format.rs]: Not sure. It's what I've been using for my own stuff. I don't feel too strongly either way, though.

Comment (SimonSapin) [inline src/format.rs]: Ok. Please change them to `.as_slice()`, then.

**Comment by sfackler** [comment]
Updated

**Comment by SimonSapin** [comment]
Thanks!

**Pull request merged**